core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java

```diff
@@ -556,6 +556,15 @@ private DiscoveryResult discoverTestClasses() {
                             enclosingClasses.put(classInfo.name(), enclosing);
                         }
                     }
+                } else if (instance.target().kind() == AnnotationTarget.Kind.FIELD) {
+                    ClassInfo classInfo = instance.target().asField().declaringClass();
+                    allTestClasses.add(classInfo.name());
+                    if (classInfo.classAnnotation(NESTED) != null) {
```

Comment:
Not sure if `@Nested` is even supported for field-style AU tests, but I guess it doesn't hurt.